src/Microsoft.AspNetCore.Identity.UI/IdentityUIConfigureStaticFiles.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Identity.UI
+{
+    internal class IdentityUIConfigureStaticFiles : IConfigureOptions<StaticFileOptions>
+        {
+            public IdentityUIConfigureStaticFiles(IHostingEnvironment environment)
+            {
+                Environment = environment;
+            }
+
+            public IHostingEnvironment Environment { get; private set; }
+
+            public void Configure(StaticFileOptions options)
+            {
+                // Basic initialization in case the options weren't initialized by any other component
+                options.ContentTypeProvider = options.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
+                if (options.FileProvider == null && Environment.WebRootFileProvider == null)
+                {
+                    throw new InvalidOperationException("Missing FileProvider.");
+                }
+
+                options.FileProvider = Environment.WebRootFileProvider;
```

Comment:
@Eilon Your comment got lost on my uber rebase.

This is just the initialization block the static files middleware has for its options. We need to duplicate it here as there is no "AddStaticFiles" service collection extension. I wanted to keep this logic separate to make explicit the gesture that I'm doing to plug our embedded file provider. This part is most likely to change in the future, so I'm not concerned about the current shape of the code.